integration/experiment/gpu_frequency_sweep/gpu_frequency_sweep.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2015 - 2022, Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+'''
+Helper functions for running gpu frequency sweep experiments.
+'''
+
+import sys
+import argparse
+import os
+
+import geopmpy.agent
+
+from experiment import launch_util
+from experiment import common_args
+from experiment import machine
+from experiment.frequency_sweep import frequency_sweep
+from experiment.uncore_frequency_sweep import uncore_frequency_sweep
+
+
+def setup_run_args(parser):
+    uncore_frequency_sweep.setup_run_args(parser)
+    parser.add_argument('--max-gpu-frequency', dest='max_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='top gpu frequency setting for the sweep')
+    parser.add_argument('--min-gpu-frequency', dest='min_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='bottom gpu frequency setting for the sweep')
+    parser.add_argument('--step-gpu-frequency', dest='step_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='increment in hertz between gpu frequency settings for the sweep')
+
+def setup_gpu_frequency_bounds(mach, min_gpu_freq, max_gpu_freq,
+                                  step_gpu_freq):
+
+    sys_min = mach.gpu_frequency_min()
+    sys_max = mach.gpu_frequency_max()
+
+    #TODO: Is there a GPU signal equivalent for this? 
```

Comment:
I don't think so. The way we did it pre-IOGroup was to request a list of allowed frequency limits from the GPU.

We might just need to pick some default number of steps to try, then get the actual applied limit from the reports.